server/src/main/java/org/elasticsearch/index/shard/IndexReaderWrapper.java

```diff
@@ -31,10 +31,9 @@
 import java.io.IOException;
 
 /**
- * Extension point to add custom functionality at request time to the {@link DirectoryReader}
- * and {@link IndexSearcher} managed by the {@link IndexShard}.
+ * Extension point to add custom functionality at request time to the {@link DirectoryReader} managed by the {@link IndexShard}.
  */
-public class IndexSearcherWrapper {
+public class IndexReaderWrapper {
```

Comment:
I wonder if we should extend this interface to be `DirectoryReader wrap(ShardId shardId, DirectoryReader reader) throws IOException` to make it clear that it's a per shard call?

Reply:
I can change the `CheckedFunction<DirectoryReader, DirectoryReader, IOException>` that I added in [link] into `CheckedBiFunction<ShardId, DirectoryReader, DirectoryReader, IOException>`. However this makes the signature a bit hard to read so maybe we can change the javadocs to make it clear that the wrapper function is called per shard ? The ShardId is not really important here.

Reply:
sounds good to me.